libraries/botbuilder-dialogs-adaptive/src/recognizers/entityRecognizers/entityRecognizer.ts

```diff
@@ -6,14 +6,107 @@
  * Licensed under the MIT License.
  */
 
-import { Entity } from 'botbuilder-core';
-import { DialogContext } from 'botbuilder-dialogs';
+import { Activity, Entity, RecognizerResult } from 'botbuilder-core';
+import { DialogContext, Recognizer } from 'botbuilder-dialogs';
+import { TextEntity } from './textEntity';
 
-export interface EntityRecognizer {
-    recognizeEntities(
+/**
+ * Entity recognizers base class.
+ */
+export class EntityRecognizer extends Recognizer {
+    /**
+     * To recognize intents and entities in a users utterance.
+     *
+     * @param {DialogContext} dialogContext Dialog Context.
+     * @param {Partial<Activity>} activity Activity.
+     * @param {object} telemetryProperties Additional properties to be logged to telemetry with event.
+     * @param {object} telemetryMetrics Additional metrics to be logged to telemetry with event.
+     * @returns {Promise<RecognizerResult>} Analysis of utterance.
+     */
+    public async recognize(
+        dialogContext: DialogContext,
+        activity: Partial<Activity>,
+        telemetryProperties?: Record<string, string>,
+        telemetryMetrics?: Record<string, number>
+    ): Promise<RecognizerResult> {
+        // Identify matched intents
+        const text = activity.text ?? '';
+
+        const recognizerResult: RecognizerResult = {
+            text,
+            intents: {},
+        };
+
+        if (!text) {
+            // nothing to recognize, return empty result
+            return recognizerResult;
+        }
+
+        const textEntity: Entity = new TextEntity(text);
+        textEntity.start = 0;
```

Comment:
It's starting to seem like these should actually be properties on the `TextEntity` class itself, and probably computed inside the constructor. What do you think?

Reply:
Yes, every time we referred `TextEntity`, we referred start/end/score properties although they are not there in the implementation of C# version. Do you think we need to expand the `TextEntity` class to have these properties? I'm not sure because we don't have these in C#.

Reply:
Seems like a fix could be made in both places, but we don't need to block on that for now. I'll file a separate issue.

Reply:
Let's just leave it unchanged here and do it in another PR?

Reply:
Yep, exactly.